scripts/make_facet_masks.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+__author__ = "Jurjen de Jong"
+
+import os
+from os.path import basename
+from argparse import ArgumentParser
+import numpy as np
+import tables
+
+from astropy.io import fits
+from casacore.tables import table
+from numba import njit, prange, set_num_threads
+from joblib import Parallel, delayed
+
+
+@njit(parallel=True)
+def add_in_place(acc, arr):
+    """
+    Adds the contents of array `arr` into `acc` element-wise in place.
+
+    Parameters:
+        acc (np.ndarray): The accumulator array.
+        arr (np.ndarray): The array to add into acc.
+    """
+    flat_size = acc.size
+    # Loop over all elements in parallel
+    for i in prange(flat_size):
+        acc.flat[i] += arr.flat[i]
+
+
+def get_data_shape(ms: str) -> tuple[int, int, int]:
+    """
+    Get the shape of the DATA column in a Measurement Set.
+
+    Parameters
+    ----------
+    ms : str
+        Path to the Measurement Set.
+
+    Returns
+    -------
+    tuple of int
+        A tuple (nrows, nfreq, npol) where:
+        - nrows is the number of rows in the table,
+        - nfreq is the number of frequency channels,
+        - npol is the number of polarisations.
+    """
+    with table(ms, ack=False) as t:
+        freq, pol = t.getdesc()["DATA"]['shape']
+        return (t.nrows(), freq, pol)
+
+
+def create_memmap(facetnumber: int | str, shape: tuple[int, int], dtype: np.dtype) -> np.memmap:
+    """
+    Create a zero-initialised memory-mapped array.
+
+    Parameters
+    ----------
+    facetnumber : int or str
+        Identifier for the facet, used in the filename.
+    shape : tuple of int
+        Dimensions of the array.
+    dtype : data-type
+        Data type of the array elements.
+
+    Returns
+    -------
+    np.memmap
+        The created memory-mapped array.
+    """
+    filename = f"FACET_{facetnumber}.dat"
+    print(f"Creating {filename}")
+    memmap_obj = np.memmap(filename, dtype=dtype, mode='w+', shape=(shape[0], shape[1]))
+    memmap_obj[:] = 0  # Initialize the file with zeros
+    return memmap_obj
+
+
+def update_memmap(dat: np.memmap, poly_number: int | str, poly_data: np.ndarray) -> None:
+    """
+    Update a memory-mapped facet file with polynomial data.
+
+    Parameters
+    ----------
+    dat : np.memmap
+        Memory-mapped array representing the facet data.
+    poly_number : int or str
+        Identifier for the polynomial data.
+    poly_data : np.ndarray
+        Array of data to be added in place.
+    """
+    facet_id = basename(dat.filename).replace("FACET_", "").replace(".dat", "")
+    if facet_id != poly_number:
+        print(f"COMPUTE {dat.filename} + POLY_{poly_number}")
+        # Get the column, convert to complex64, and add it in place
+        add_in_place(dat, poly_data)
+
+
+def add_axis(arr: np.ndarray, ax_size: int) -> np.ndarray:
+    """
+    Add a new axis to an array with a given size.
```

Comment:
The new axis is populated with the values of the old array. It would be good to mention this.